src/components/viewContainer.js

```diff
@@ -36,53 +36,56 @@ import 'css!components/viewManager/viewContainer';
-                }
+            if (currentPage) {
+                triggerDestroy(currentPage);
+            }
 
-                let view = newView;
+            let view = newView;
 
-                if (typeof view == 'string') {
-                    view = document.createElement('div');
-                    view.innerHTML = newView;
-                }
+            if (typeof view == 'string') {
+                view = document.createElement('div');
+                view.innerHTML = newView;
+            }
 
-                view.classList.add('mainAnimatedPage');
+            view.classList.add('mainAnimatedPage');
 
-                if (currentPage) {
-                    if (newViewInfo.hasScript && window.$) {
-                        mainAnimatedPages.removeChild(currentPage);
-                        view = $(view).appendTo(mainAnimatedPages)[0];
-                    } else {
-                        mainAnimatedPages.replaceChild(view, currentPage);
-                    }
+            if (currentPage) {
+                if (newViewInfo.hasScript && window.$) {
+                    mainAnimatedPages.removeChild(currentPage);
+                    view = $(view).appendTo(mainAnimatedPages)[0];
                 } else {
-                    if (newViewInfo.hasScript && window.$) {
-                        view = $(view).appendTo(mainAnimatedPages)[0];
-                    } else {
-                        mainAnimatedPages.appendChild(view);
-                    }
+                    mainAnimatedPages.replaceChild(view, currentPage);
                 }
-
-                if (options.type) {
-                    view.setAttribute('data-type', options.type);
+            } else {
+                if (newViewInfo.hasScript && window.$) {
+                    view = $(view).appendTo(mainAnimatedPages)[0];
+                } else {
+                    mainAnimatedPages.appendChild(view);
                 }
+            }
 
-                const properties = [];
+            if (options.type) {
+                view.setAttribute('data-type', options.type);
+            }
 
-                if (options.fullscreen) {
-                    properties.push('fullscreen');
-                }
+            const properties = [];
 
-                if (properties.length) {
-                    view.setAttribute('data-properties', properties.join(','));
-                }
+            if (options.fullscreen) {
+                properties.push('fullscreen');
+            }
 
-                allPages[pageIndex] = view;
-                setControllerClass(view, options).then(() => {
+            if (properties.length) {
+                view.setAttribute('data-properties', properties.join(','));
+            }
+
+            allPages[pageIndex] = view;
+
+            return setControllerClass(view, options)
+                // Timeout for polyfilled CustomElements (webOS 1.2)
+                .then(() => new Promise((resolve) => setTimeout(resolve, 0)))
```

Comment:
This seems pretty hacky. Is there not a better way to delay until the polyfill is ready?

Reply:
Need to wait for upgrade of elements (not the polyfill itself). Didn't find anything good.

Without timeout, it looks like:
```
set innerHTML/appendChild
call view constructor
    access to a not yet upgraded element
MutationObserver upgrades elements
```
With timeout, it is presumably:
```
set innerHTML/appendChild
schedule call of view constructor via zero timeout
MutationObserver upgrades elements
[timeout] call view constructor
    access to an upgraded element
```

As I said, we can use `window.CustomElements.upgradeSubtree(view, true)` (`true` to call `attachedCallback`; _tried - works for current issue_):
```js
return setControllerClass(view, options)
    .then(() => {
        // Polyfilled CustomElements (webOS 1.2, webOS 2, Firefox 59+) is implemented
        // via MutationObserver. Therefore, it takes time to upgrade the added elements.
        // But we can just force it to upgrade.
        window.CustomElements.upgradeSubtree(view, true);

        if (onBeforeChange) {
            onBeforeChange(view, false, options);
        }
```

**Another option**: add timeouts or `upgradeSubtree` in place.
https://github.com/jellyfin/jellyfin-web/blob/2e6211ce31b3966c19c461b9269d9478f87b08e4/src/components/subtitlesync/subtitlesync.js#L24-L27
_This is the only (known) place where Firefox fails because it also uses `webcomponents.js`_